tools/templates/veneer_debug_symbols.template

```diff
@@ -0,0 +1,5 @@
+{% for partition in partitions %}
+{% for function in partition.manifest.secure_functions%}
+-Wl,--undefined={{function.signal.lower()}}
```

Comment:
After reading the docs for --undefined I am surprised that this did not bloat the binary.

Perhaps the binary that was tested already had all veneers?

Can you test/confirm that we can have a veneer in this list without it ending up in the tfm_s.efl?

Reply:
This doesn't add the symbols to the build because whatever is preventing these from being optimized out is already applied.

So there is no risk of bloat we believe.

Reply:
I tested it with `CONFIG_TFM_CMAKE_BUILD_TYPE_DEBUG=y` and without it, hex size is not changing. 

The rest was discussed offline